core/app/user/application/use_cases/update_user.py

```diff
@@ -0,0 +1,30 @@
+from uuid import UUID
+from core.app.user.domain.ports import UserRepository
+from core.app.user.domain.entities import User
+from core.app.user.application.exceptions import UserNotFoundException
+from .dto import UpdateUserDTO
+from pydantic import ValidationError
+
+class UpdateUserUseCase:
+    def __init__(self, user_repository: UserRepository):
+        self.user_repository = user_repository
+
+    def execute(self, user_id: UUID, user_data: dict) -> User:
+        user = self.user_repository.get(user_id)
+        if not user:
+            raise UserNotFoundException("User not found")
```

Comment:
Be more specific on the error messages
```suggestion
        if not user:
            raise UserNotFoundException(f"User not with id {user_id} not found")
```